x/datapool/keeper/pool.go

```diff
@@ -0,0 +1,49 @@
+package keeper
+
+import (
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	"github.com/medibloc/panacea-core/v2/types/assets"
+	"github.com/medibloc/panacea-core/v2/x/datapool/types"
+)
+
+func (k Keeper) RegisterDataValidator(ctx sdk.Context, dataValidatorAddress sdk.AccAddress, dataValidator types.DataValidator) error {
+	store := ctx.KVStore(k.storeKey)
+	dataValidatorKey := types.GetKeyPrefixDataValidator(dataValidatorAddress)
+	if store.Has(dataValidatorKey) {
+		return sdkerrors.Wrapf(types.ErrDataValidatorAlreadyExist, "the data validator %s is already exists", dataValidatorAddress)
+	}
+
+	balance := k.bankKeeper.GetBalance(ctx, dataValidatorAddress, assets.MicroMedDenom)
+	if balance.IsZero() {
+		return sdkerrors.ErrInsufficientFunds
+	}
```

Comment:
My guess is that there will be cases where MED is 0, although it is a valid validator address.
How about checking that the account exists and that the PublicKey exists in the account?

Reply:
Ah yes. We talked about it. I'll change it not only for this, but also for test code.